src/agent/rustjail/src/mount.rs

```diff
@@ -668,6 +668,44 @@ fn parse_mount(m: &Mount) -> (MsFlags, String) {
     (flags, data.join(","))
 }
 
+fn secure_join(rootfs: &str, unsafe_path: &str) -> String {
```

Comment:
- Please add a comment explaining what this function does.
- Please can you add a set of unit tests for this function that test all the awkward boundary conditions (`rootfs == ""`, `unsafe_path == ""`, *etc*). I'd probably use a table-driven set of tests as documented in https://github.com/kata-containers/tests/blob/master/Unit-Test-Advice.md#rust-table-driven-tests.
- There are lots of calls to `unwrap()` in this fuction which makes me wonder if we want to return `Result<String>` here.

Reply:
OK. got it

Reply:
Updated